src/quokka_editor_back/tests/routers/test_websockets.py

```diff
@@ -0,0 +1,317 @@
+        mocked_manager.broadcast.assert_called_once_with(
+            message=json.dumps(
+                {
+                    "message": f"User {token} Disconnected from the file.",
+                    "user_token": token,
+                }
+            ),
+            websocket=websocket,
+            document_id=document.id,
+            user_token=token,
+            send_to_owner=False,
+        )
+        close_mock.assert_called_once()
+        mocked_manager.disconnect.assert_called_once_with(websocket, document.id)
+
+
+@pytest.mark.skip("Testing the websocket connection will be fixed later")
+@patch("quokka_editor_back.actors.transform_document", new_callable=AsyncMock)
+@patch("quokka_editor_back.routers.websockets.get_redis", new_callable=AsyncMock)
+@patch(
+    "quokka_editor_back.routers.websockets.process_websocket_message",
+    side_effect=Exception("Message"),
+)
+async def test_websocket_with_disconnect_error(
+    mocked_process_websocket_message,
+    mocked_get_redis,
+    mocked_transform_document,
+    client: TestClient,
+    websocket,
+    active_user: UserLogin,
+    document: Document,
+    mocker,
+):
+    # Given
+    token = auth_handler.encode_token(active_user.username)
+    mock_manager = AsyncMock(spec=manager)
+
+    # When
+    with client.websocket_connect(f"/ws/{document.id}?token={token}"):
+        mocked_manager = mocker.patch(
+            "quokka_editor_back.routers.websockets.manager", new_callable=mock_manager
+        )
+
+        mocked_manager.connect.assert_called_once_with(mocker.Mock(), document.id)
+        mocked_manager.disconnect.assert_called_once_with(mocker.Mock(), document.id)
+
+
+@pytest.mark.skip("Testing the websocket connection will be fixed later")
+async def test_websocket_authentication(
+    client: TestClient, websocket, active_user: UserLogin, document: Document, mocker
+):
+    # Given
+    token = auth_handler.encode_token(active_user.username)
+    mock_authenticate_websocket = mocker.patch(
+        "quokka_editor_back.routers.websockets.authenticate_websocket",
+        new_callable=AsyncMock(return_value=(123, ShareRole.EDIT)),
+    )
+
+    # When
+    with client.websocket_connect(f"/ws/{document.id}?token={token}"):
+        # Then
+        mock_authenticate_websocket.assert_called_once_with(document.id, token)
+
+
+@pytest.mark.skip("Testing the websocket connection will be fixed later")
+async def test_websocket_communication(
+    client: TestClient, active_user: UserLogin, document: Document, mocker
+):
+    # Given
+    AsyncMock(spec=WebSocket)
+    mocked_manager_broadcast = mocker.patch(
+        "quokka_editor_back.routers.websockets.manager.broadcast", AsyncMock()
+    )
+    token = auth_handler.encode_token(active_user.username)
+
+    # When
+    with client.websocket_connect(f"/ws/{document.id}?token={token}") as websocket:
+        # Then
+        # mock_websocket.receive_text.assert_called_once_with()
```

Comment:
we are not checking it? if not remove this comment

Reply:
These tests for websocket don't work